functions/github/github.js

```diff
@@ -0,0 +1,258 @@
+          },
+          per_page: 100,
+        });
+
+    return checkRuns;
+  } catch (err) {
+    error(
+        `Error fetching check suites on ${ref} from GitHub:`,
+        {error: err.message},
+    );
+    throw err;
+  }
+}
+
+/**
+ * Fetches and parses release configuration data from a GitHub repository.
+ *
+ * @param {Octokit} octokit The authenticated Octokit client.
+ * @param {Object} releaseData The release data containing the branch name.
+ * @return {Promise<Object>} A promise that resolves to the parsed configuration
+ * data.
+ */
+async function getReleaseConfig(octokit, releaseData) {
+  const releaseConfigJSON = await getRepositoryContent(
+      octokit, releaseData.releaseBranchName, "release.json",
+  );
+  const releaseConfig = JSON.parse(releaseConfigJSON);
+  return releaseConfig;
+}
+
+/**
+ * Fetches and parses release report data from a GitHub repository.
+ *
+ * @param {Octokit} octokit The authenticated Octokit client.
+ * @param {Object} releaseData The release data containing the branch name.
+ * @return {Promise<boolean>} A promise that resolves to the parsed report
+ * data.
+ */
+async function getReleaseReport(octokit, releaseData) {
+  const releaseReportJSON = await getRepositoryContent(
+      octokit, releaseData.releaseBranchName, "release_report.json",
+  );
+  const releaseReport = JSON.parse(releaseReportJSON);
+  return releaseReport;
+}
+
+/**
+ * Fetches and returns metadata for each library in the release configuration.
+ * The metadata includes the updated version, whether the library is opted in
+ * to the release, and whether the library is lockstep. A library is lockstep
+ * if it is part of the release but has no changes. A library is opted in if
+ * it was not originally part of the release report, but has been manually opted
+ * in by being added to the release configuration.
+ *
+ * @param {Octokit} octokit - The authenticated Octokit client.
+ * @param {string} releaseBranchName - The release branch name
+ * @param {Object} releaseReport - The release report
+ * @param {Object} releaseConfig - The release configuration with the processed
+ * library names.
+ * @return {Promise<Object>} A promise that resolves to an object mapping
+ * library names to metadata.
+ */
+async function getLibraryMetadata(
+    octokit,
+    releaseBranchName,
+    releaseReport,
+    releaseConfig,
+) {
+  const libraryMetadata = {};
+  const libraryVersions = await getLibraryVersions(
+      octokit,
+      releaseBranchName,
+      releaseConfig,
+  );
+  log("Fetched library versions", {libraryVersions: libraryVersions});
+
+  for (const library in libraryVersions) {
+    if (Object.prototype.hasOwnProperty.call(libraryVersions, library)) {
+      libraryMetadata[library] = {
+        "updatedVersion": libraryVersions[library],
+        "optedIn": !releaseReport.changesByLibraryName[library],
+        "isLockstep": !releaseReport.changesByLibraryName[library] ||
```

Comment:
perhaps renaming this to optedInByLibraryGroup? Or just noChanges? 

isLockstep isn't a bad name, its just not one we use anywhere else, so it might be confusing here

Reply:
I feel like `optedInByLibraryGroup` would be the most accurate, will change.
I was unsure about `isLockstep`, what do we usually call this?